### ISSUE TITLE
Missing gradient in AdaGrad update rule?

### ISSUE THREAD

**Issue by ghost** [PyMLVizard/PyMLViz#3]
I'm not certain, but I believe in the update rule for the AdaGrad gradient descent method the gradient is missing after the adjusted learning rate.

I think in the notebook `GradientDescent_Variants.ipynb`,
```
$$ \\theta_{i,n+1} = \\theta_{i,n} - \\frac{\\lambda}{\\sqrt{g_{i,n}^2 + \\epsilon}} $$
```
should look like
```
$$ \\theta_{i,n+1} = \\theta_{i,n} - \\frac{\\lambda}{\\sqrt{g_{i,n}^2 + \\epsilon}} \frac{\partial}{\partial \theta_{i,n}} \mathcal{L}(\mathbf{\theta}_n) $$
```

I compared this to [the section in the Wikipedia article on stochastic gradient descent](https://en.wikipedia.org/wiki/Stochastic_gradient_descent#AdaGrad), where `$g_j$` is multiplied after the adjusted learning rate.

**Comment by bertschi** [comment]
Thanks for the catch. Is fixed in master now.
